common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbDate.java

```diff
@@ -488,18 +488,55 @@ public static long parse(String value) {
     }
 
     private static Instant parseInstant(String s) {
+        DateTimeFormatter formatter;
+        boolean isIsoFormat = s.length() > 0 && Character.isDigit(s.charAt(0));
```

Comment:
I don't like this approach at all. Better just create a list of DateTimeFormatter's that you would like to support and try them one by one. Catch the [DateTimeParseException](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeParseException.html) and not all exceptions.